src/persist/src/storage.rs

```diff
@@ -141,14 +141,61 @@ pub enum Atomicity {
     AllowNonAtomic,
 }
 
-/// An abstraction over a `bytes key`->`bytes value` store.
+/// An abstraction over read-only access to a `bytes key`->`bytes value` store.
+#[async_trait]
+pub trait BlobRead: Send + 'static {
+    /// Returns a reference to the value corresponding to the key.
+    async fn get(&self, key: &str) -> Result<Option<Vec<u8>>, Error>;
+
+    /// List all of the keys in the map.
+    async fn list_keys(&self) -> Result<Vec<String>, Error>;
+
+    /// Synchronously closes the blob, causing all future commands to error.
+    ///
+    /// If `Self` also implements Read, this releases the exclusive-writer lock.
```

Comment:
did you mean "also implements Blob" here?

Reply:
yup!